### PR TITLE
E2241 Heatgrid fixes and improvements

### PR DESCRIPTION
**Pull request by EshwarCVS**
E2241/Heatgrid fixes and improvements/Issue#1869

Changes made:

- Removed countofcomments attribute from vm_question_response_row.rb file.
- Added metric_hash attribute to store the metric values for each response in vm_question_response_row.rb.
- Iterate values in metric_hash in view_team.html.erb file to show the metric values in the heat grid.
- Added a new method calculate_metrics, that calls all other methods for different metric calculations.

Mentioning the contributors as we have done peer programming:
@EshwarCVS
@gokul-koganti
@J-Sumer
@sbosenc

### CODE REVIEW

File: app/models/vm_question_response.rb [hunk 1]
```diff
@@ -161,12 +161,32 @@ def add_answer(answer)
...
+    number_of_comments_greater_than_20_words
+  end
+
   def number_of_comments_greater_than_10_words
     @list_of_reviews.each do |review|
       answers = Answer.where(response_id: review.response_id)
       answers.each do |answer|
         @list_of_rows.each do |row|
-          row.countofcomments = row.countofcomments + 1 if row.question_id == answer.question_id && answer.comments && answer.comments.split.size > 10
+          row.metric_hash["countofcomments"] = 0 if row.metric_hash["countofcomments"] == nil
```
Comment (johnbumgardner) [inline app/models/vm_question_response.rb:174]:
```suggestion
          row.metric_hash["countofcomments"] = 0 if row.metric_hash["countofcomments"].nil?
```

File: app/models/vm_question_response.rb [hunk 2]
```diff
@@ -161,12 +161,32 @@ def add_answer(answer)
...
+  end
+
+  # Incase if new metirc is added
+
+  def number_of_comments_greater_than_20_words
+    @list_of_reviews.each do |review|
+      answers = Answer.where(response_id: review.response_id)
+      answers.each do |answer|
+        @list_of_rows.each do |row|
+          row.metric_hash["countofcomments20"] = 0 if row.metric_hash["countofcomments20"] == nil
```
Comment (johnbumgardner) [inline app/models/vm_question_response.rb:188]:
```suggestion
          row.metric_hash["countofcomments20"] = 0 if row.metric_hash["countofcomments20"].nil?
```

File: app/views/grades/view_team.html.erb [hunk 3]
```diff
@@ -106,6 +106,10 @@
 
                 fix comment end
               -->
+              <br/>
+        <% vm.list_of_rows[0].metric_hash.each do |key, value| %>
```
Comment (johnbumgardner) [inline app/views/grades/view_team.html.erb:110]:
```suggestion
        <% vm.list_of_rows.first.metric_hash.each do |key, value| %>
```

File: app/views/grades/view_team.html.erb [hunk 4]
```diff
@@ -127,9 +131,16 @@
...
                 Avg<span></span>
               </th>
-              <th class="sorter-false">
-                <span  data-toggle="tooltip" data-placement="right" title="<%=t(".metric_tooltip")%>"><%=t ".metric1"%></span>
-              </th>
+
+                <!-- <span  data-toggle="tooltip" data-placement="right" title="<%=t(".metric_tooltip")%>"><%=t ".metric1"%></span> -->
+              
+
+              <% vm.list_of_rows[0].metric_hash.each do |key, value| %>
```
Comment (johnbumgardner) [inline app/views/grades/view_team.html.erb:138]:
```suggestion
              <% vm.list_of_rows.first.metric_hash.each do |key, value| %>
```

**Review by johnbumgardner — CHANGES_REQUESTED** [review]
make changes

File: app/models/vm_question_response.rb (hunk 1 above)
Comment (J-Sumer) [reply app/models/vm_question_response.rb]: Changes are done.

File: app/models/vm_question_response.rb (hunk 2 above)
Comment (J-Sumer) [reply app/models/vm_question_response.rb]: Changes are done.

File: app/views/grades/view_team.html.erb (hunk 3 above)
Comment (J-Sumer) [reply app/views/grades/view_team.html.erb]: Changes are done.

File: app/views/grades/view_team.html.erb (hunk 4 above)
Comment (J-Sumer) [reply app/views/grades/view_team.html.erb]: Changes are done.

File: app/models/vm_question_response.rb [hunk 5]
```diff
@@ -161,31 +161,33 @@ def add_answer(answer)
...
 
+  # This method is responsible for checking whether a review comment contians more than 10 words.
   def number_of_comments_greater_than_10_words
     @list_of_reviews.each do |review|
       answers = Answer.where(response_id: review.response_id)
       answers.each do |answer|
         @list_of_rows.each do |row|
-          row.metric_hash["countofcomments"] = 0 if row.metric_hash["countofcomments"] == nil
-          row.metric_hash["countofcomments"] = row.metric_hash["countofcomments"] + 1 if row.question_id == answer.question_id && answer.comments && answer.comments.split.size > 10
+          row.metric_hash["comments #10"] = 0 if row.metric_hash["comments #10"].nil?
```
Comment (nnhimes) [inline app/models/vm_question_response.rb]: Change string to "> 10 Word Comments" maybe, or at least capitalize what you have. Its not obvious what this means

**Comment by nnhimes** [comment]
Fix rubocop and linting issues

**Pull request merged**